test/language/literals/string/legacy-non-octal-escape-sequence-strict.js

```diff
@@ -0,0 +1,36 @@
+// Copyright (C) 2017 Kevin Gibbons. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-additional-syntax-string-literals
+es6id: B1.2
+description: >
+    LegacyOctalEscapeSequence is not enabled in strict mode code (regardless of
+    the presence of Annex B)
+info: >
+    EscapeSequence ::
+      CharacterEscapeSequence
+      LegacyOctalEscapeSequence
+      HexEscapeSequence
+      UnicodeEscapeSequence
+
+    LegacyOctalEscapeSequence ::
+      OctalDigit [lookahead ∉ OctalDigit]
+      ZeroToThree OctalDigit [lookahead ∉ OctalDigit]
+      FourToSeven OctalDigit
+      ZeroToThree OctalDigit OctalDigit
+
+    ZeroToThree :: one of
+      0 1 2 3
+
+    FourToSeven :: one of
+      4 5 6 7
+
+    This definition of EscapeSequence is not used in strict mode or when
+    parsing TemplateCharacter.
+flags: [onlyStrict]
+negative:
+  phase: early
+  type: SyntaxError
+---*/
+
+'\08';
```

Comment:
This is a very particular case, banned because `\0` has a no decimal digit lookahead, in combination with how it would've been handled as a legacy octal literal because 8 isn't an octal digit (right?). Maybe it'd be worth adding another test that's a little more normal, such as `\12`.